File: OOP/tic-top-toe-app/src/com/techlab/board/Board.java
package com.techlab.board;

import com.techlab.cell.Cell;
import com.techlab.enums.Mark;
import com.techlab.exceptions.CellIsAlredyMarkedException;

public class Board {
	private Cell[] cellsArray;
	private static int NUMBER_OF_CELL = 9;

	public Board() {
		cellsArray = new Cell[NUMBER_OF_CELL];
		for (int index = 0; index < NUMBER_OF_CELL; index++)
			cellsArray[index] = new Cell();
	}

	public void putMarkOnCell(Mark mark, int cellNumber) throws CellIsAlredyMarkedException {
		cellsArray[cellNumber].setCellState(mark);
	}
	
	public Mark getMarkOnCell(int cellNumber) {
		return cellsArray[cellNumber].getCellState();
	}
	
	public Cell[] getCellsArray() {
		return cellsArray;
	}
		
	public boolean isFilledOut(){
		for (int index = 0; index < NUMBER_OF_CELL; index++) {
			if(this.getMarkOnCell(index) != Mark.EMPTY) {
				return true;
			}
		}
		return false;
	}
}

File: nodejs/tic-tac-toe/board.js
let Cell = require('./cell');

class Board {
    constructor() {
        this.cells = [];
        for (let index = 0; index < 9; index++) {
            let cell = new Cell();
            this.cells.push(cell);
        }
    }

    markOnCell(cellNumber, mark) {
        this.cells[cellNumber].putMark(mark);
    }

    getMarkOnCell(cellNumber) {
        return this.cells[cellNumber].getMark();
    }

    printBoard() {
        for (let i = 0; i < 9; i += 3) {
            let row = "";
            for (let j = i; j < 3 + i && j < 9; j++) {
                row += this.cells[j].getMark() + "\t";
            }
            console.log(row);
        }
    }
}

module.exports = Board;